### PR TITLE
Reland "Update PopupMenuButton to match Material Design spec"

### LINKED ISSUE #6107: The position of PopupMenuButton's menu does not follow the Material Design spec
In the Menus demo of the Gallery:
1. Tap 'An item with a simple menu'.
2. Tap 'Menu item value three'.
3. Tap 'An item with a simple menu' item again.

Expected: menu opens up from the bottom (where you're actually tapping)
Observed: menu opens up from the top

### LINKED ISSUE #69100: The position of PopupMenuBottom's menu is incorrect.
the PR(#64678) caused this bug.

## Steps to Reproduce

1. run project [bug.zip](https://github.com/flutter/flutter/files/5444871/bug.zip)
2. click PopupMenuButton in AppBar

**Expected results:** 
the top of the menu and PopupMenuButton are aligned.

**Actual results:** 
On the windows platform, the display is correct.
On the android platform, the menu is offset downward.

screenshot on windows:
Image

screenshot on android:
Image

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, 1.24.0-3.0.pre, on Microsoft Windows [Version 10.0.19041.572], locale zh-CN)
    • Flutter version 1.24.0-3.0.pre at C:\SDK\flutter
    • Framework revision 2783f8e2e1 (5 days ago), 2020-10-22 09:36:06 -0700
    • Engine revision defa8be2b1
    • Dart version 2.11.0 (build 2.11.0-242.0.dev)
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\SDK\android-sdk
    • Platform android-30, build-tools 29.0.3
    • ANDROID_HOME = C:\SDK\android-sdk
    • Java binary at: C:\SDK\android-studio-beta\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.4.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.4.29613.14
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\SDK\android-studio-beta
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2020.2)
    • IntelliJ at C:\Users\ipcjs\AppData\Local\JetBrains\Toolbox\apps\IDEA-C\ch-0\202.4357.23
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.50.1)
    • VS Code at C:\Users\ipcjs\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.15.1

[✓] Proxy Configuration
    • HTTP_PROXY is set
    • NO_PROXY is localhost,127.0.0.1,dl.google.com,maven.aliyun.com,cdn.npm.taobao.org,registry.npm.taobao.org,storage.flutter-io.cn,pub.flutter-io.cn,services.gradle.org,apple.com,downloads.gradle-dn.com
    • NO_PROXY contains 127.0.0.1
    • NO_PROXY contains localhost

[✓] Connected device (5 available)
    • SM G960U (mobile) • 433943354c313098 • android-arm64  • Android 10 (API 29)
    • Windows (desktop) • windows          • windows-x64    • Microsoft Windows [Version 10.0.19041.572]
    • Web Server (web)  • web-server       • web-javascript • Flutter Tools
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 86.0.4240.111
    • Edge (web)        • edge             • web-javascript • Microsoft Edge 86.0.622.51

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by xu-baolin**
Relands #70160

### New change

Use a common method of getting semantic nodes like other test cases, instead of depending on the structure of the widget tree.
CC @gspencergoog 

### Related test
'Popup Menu has correct Android semantics'

Fixes #6107 
Fixes #69100

### CODE REVIEW

**Comment by xu-baolin** [comment]
Now the failed test can be passed.
image

File: dev/integration_tests/android_semantics_testing/test_driver/main_test.dart [hunk 1]
```diff
@@ -432,20 +438,21 @@ void main() {
...
-                  isEnabled: true,
-                  isFocusable: true,
-                  actions: <AndroidSemanticsAction>[
-                    if (item == popupItems.first) AndroidSemanticsAction.clearAccessibilityFocus,
-                    if (item != popupItems.first) AndroidSemanticsAction.accessibilityFocus,
-                    AndroidSemanticsAction.click,
-                  ],
-                ),
-                reason: "Popup $item doesn't have the right semantics");
+              await getSemantics(find.byValueKey('$popupKeyValue.$item')),
```
Comment (xu-baolin) [inline dev/integration_tests/android_semantics_testing/test_driver/main_test.dart:441]: Here is just an indented alignment.

**Review by Piinks — APPROVED** [review]
Flutter_LGTM
Thanks for updating this to reland it! :)

**Comment by Piinks** [comment]
This appears to be breaking a lot of internal tests that were not executed as part of the sampling of tests that were run in the `Google testing` check, revert is in #75338

**Pull request merged**